src/test/java/com/microsoft/azure/spring/data/documentdb/core/DocumentDbTemplateIT.java

```diff
@@ -85,50 +88,66 @@ public void cleanup() {
     }
 
     @Test(expected = RuntimeException.class)
-    public void testInsertDuplicateId() throws Exception {
+    public void testInsertDuplicateId() {
         dbTemplate.insert(Person.class.getSimpleName(), TEST_PERSON, null);
     }
 
     @Test
     public void testFindAll() {
-        final List<Person> result = dbTemplate.findAll(Person.class.getSimpleName(), Person.class, null, null);
+        final List<Person> result = dbTemplate.findAll(Person.class.getSimpleName(), Person.class);
         assertThat(result.size()).isEqualTo(1);
         assertThat(result.get(0)).isEqualTo(TEST_PERSON);
     }
 
     @Test
-    public void testFindAllPartition() {
+    public void testFindAllByPartition() {
         setupPartition();
 
-        final List<Person> result = dbTemplate.findAll(Person.class.getSimpleName(),
-                Person.class, PARTITION_KEY, TEST_PERSON.getLastName());
+        final Criteria criteria = new Criteria("lastName");
+        criteria.is(TEST_PERSON.getLastName());
+        final Query query = new Query(criteria);
 
+        final List<Person> result = dbTemplate.find(query, Person.class, Person.class.getSimpleName());
         assertThat(result.size()).isEqualTo(1);
-        assertThat(result.get(0)).isEqualTo(TEST_PERSON);
+        assertTrue(result.get(0).equals(TEST_PERSON));
     }
 
     @Test
     public void testFindById() {
         final Person result = dbTemplate.findById(Person.class.getSimpleName(),
-                TEST_PERSON.getId(), Person.class, null);
+                TEST_PERSON.getId(), Person.class);
         assertTrue(result.equals(TEST_PERSON));
 
         final Person nullResult = dbTemplate.findById(Person.class.getSimpleName(),
-                TEST_NOTEXIST_ID, Person.class, null);
+                TEST_NOTEXIST_ID, Person.class);
         assertThat(nullResult).isNull();
     }
 
     @Test
-    public void testFindByIdPartition() {
+    public void testFindByIdWithPartition() {
         setupPartition();
 
-        final Person result = dbTemplate.findById(Person.class.getSimpleName(),
-                TEST_PERSON.getId(), Person.class, TEST_PERSON.getLastName());
-        assertTrue(result.equals(TEST_PERSON));
+        final Criteria criteria = new Criteria("id");
+        criteria.is(TEST_PERSON.getId());
+        criteria.and("lastName").is(TEST_PERSON.getLastName());
+        final Query query = new Query(criteria);
 
-        final Person nullResult = dbTemplate.findById(Person.class.getSimpleName(),
-                TEST_NOTEXIST_ID, Person.class, TEST_PERSON.getLastName());
-        assertThat(nullResult).isNull();
+        final List<Person> result = dbTemplate.find(query, Person.class, Person.class.getSimpleName());
+        assertThat(result.size()).isEqualTo(1);
+        assertTrue(result.get(0).equals(TEST_PERSON));
+    }
+
+    @Test
+    public void testFindByNonExistIdWithPartition() {
```

Comment:
might need to consider move partition related tests into seperate test class.

Reply:
Split the test into 2 test classes [here](https://github.com/Microsoft/spring-data-documentdb/pull/37/commits/de7076b777bcc8ef318ef02d30a444fc7d332614).